transport/grpc/inbound.go

```diff
@@ -81,6 +66,55 @@ func (i *inbound) Start(service transport.ServiceDetail, d transport.Deps) error
 	return nil
 }
 
+/*
+ServiceDescs in GRPC define how clients can interact with a server and usually have well defined schemas to match
+request methods with service functions.  For our purposes we are avoiding this check by dynamically creating the
+serviceDescs from the YARPC registered procedures.  This allows us to see what methods are available and to create an
+individual routing mechanism to each one.  All the routes will be forwarded to the same "Handler" which will do YARPC
+routing to handle methods and encodings.
+*/
+func getServiceDescs(gHandler handler) *[]grpc.ServiceDesc {
+	var serviceDescs []grpc.ServiceDesc
+
+	// Create separate routes for each service & procedure
+	for service, procs := range getServiceProcedures(gHandler.Registry) {
+		serviceDescs = append(serviceDescs, *createServiceDesc(gHandler, service, procs))
+	}
+
+	// If no service procedures are configured insert a default serviceProcedure
+	if len(serviceDescs) == 0 {
+		serviceDescs = append(serviceDescs, *createServiceDesc(gHandler, "yarpc", []string{"yarpc"}))
+	}
+
+	return &serviceDescs
+}
+
+func getServiceProcedures(reg transport.Registry) map[string][]string {
+	// Go through the registry to find all the methods that are currently attached to inbounds
+	serviceProcedures := make(map[string][]string)
+	for _, sp := range reg.ServiceProcedures() {
+		serviceProcedures[sp.Service] = append(serviceProcedures[sp.Service], sp.Procedure)
+	}
+	return serviceProcedures
+}
+
+func createServiceDesc(gHandler handler, service string, procedures []string) *grpc.ServiceDesc {
+	methodDescs := make([]grpc.MethodDesc, 0, len(procedures))
+	for _, proc := range procedures {
+		methodDescs = append(methodDescs, grpc.MethodDesc{
+			MethodName: url.QueryEscape(proc),
+			Handler:    gHandler.Handle, // Catchall method that does custom YARPC routing
```

Comment:
Nit: When the new slice is a 1-1 mapping of another slice, it's slightly more readable to do it without appends:

```
methodDescs := make([]grpc.MethodDesc, len(procedures))
for i, proc := range procedures {
  methodDescs[i] = grpc.MethodDesc{
    MethodName: url.QueryEscape(proc),
    Handler: grpc.Handle,
  }
}
```